### PR TITLE
Change 'ancestor_of'/'descendant_of' to 'with_descendants'/'with_ancestors'

### LINKED ISSUE #2209: Replace `ancestor_of` and `descendant_of` QueryBuilder join specifications
In accordance with #2183, the QueryBuilder relationship indicators "ancestor_of" and "descendant_of" need to be updated to "with_descendant" and "with_ancestor", respectively.

### PR DESCRIPTION
**Pull request by ConradJohnston**
Fixes #2209 . 

Deprecates the Querybuilder 'ancestor_of'/'descendant of' join type specifications and replaces them with 'with_descendants'/'with_ancestors' respectively. All deprecated uses are updated and the docs changed to reflect this new convention.

### CODE REVIEW

File: aiida/orm/querybuilder.py [hunk 1]
```diff
@@ -1659,7 +1661,7 @@ def _build(self):
             isouterjoin = verticespec.get('outerjoin')
             edge_tag = verticespec['edge_tag']
 
-            if verticespec['joining_keyword'] in ('descendant_of', 'ancestor_of'):
+            if verticespec['joining_keyword'] in ('with_ancestors', 'with_descendants'):
```
Comment (giovannipizzi) [inline aiida/orm/querybuilder.py]: If the user uses the old 'descendant_of', will this still work? Maybe we should keep all four options for now?

File: aiida/orm/querybuilder.py [hunk 2]
```diff
@@ -1262,7 +1262,7 @@ def _join_descendants_recursive(self, joined_entity, entity_to_join, isouterjoin
         """
 
         self._check_dbentities((joined_entity, self._impl.Node), (entity_to_join, self._impl.Node),
-                               'descendant_of_beta')
+                               'with_ancestors_beta')
```
Comment (giovannipizzi) [inline aiida/orm/querybuilder.py]: Is this `_beta` version still implemented? Probably @lekah can answer this

Comment (lekah) [reply aiida/orm/querybuilder.py]: No, what we're doing now (transitive closure on the fly) used to be called _beta. This string passed is just for the exception, and can be changed without a problem (i.e. remove _beta in the string).
My suggest: without a new issue within this pull request, if noone objects.

Comment (giovannipizzi) [reply aiida/orm/querybuilder.py]: Good for me!

Comment (ConradJohnston) [reply aiida/orm/querybuilder.py]: Made these changes.

File: aiida/orm/querybuilder.py (hunk 1 above)
Comment (ConradJohnston) [reply aiida/orm/querybuilder.py]: Yes, well spotted. Put these back in.

**Pull request merged**